Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -154,12 +156,62 @@ protected override void StopScanNative()
             }
         }
 
-        protected override Task ConnectToDeviceNativeAsync(IDevice device, bool autoconnect, CancellationToken cancellationToken)
+        protected override Task ConnectToDeviceNativeAsync(IDevice device, bool autoconnect, CancellationToken cancellationToken, bool forceBleTransport)
         {
             AddToDeviceOperationRegistry(device);
-            ((BluetoothDevice)device.NativeDevice).ConnectGatt(Application.Context, autoconnect, _gattCallback);
+
+            #region isLeBrErdCapableHost : Possible default decision for Simultaneous LE and BR/EDR to Same Device Capable -> DualMode
+
+            //sometimes with dual mode devices, we have to use the LE transport in the connectGatt method.
+            //otherwise the device will not respond to a create connection request (legacy)
+            //the device will only respond to a create LE connection
+            //by default the transport mode is AUTO and in most case Android will decide to use BR/EDR connection (legacy)
+            //the following is a proposition of how handle it by default.
+            //if the device advertise for the 'Simultaneous LE and BR/EDR to Same Device Capable' capability (DualMode)
+            //the default transport mode should be LE in most of context
+            //For now, just add a boolean to force BLE transport in connectToDevice methods
+
+            //bool isLeBrErdCapableHost = nativeDevice.Type == BluetoothDeviceType.Dual;
```

Comment:
So the purpose of isLeBrErdCapableHost is to force a Dual device to always take into account this additional parameter. 
Without this flag an if the parameter is left to it's default value (false) a Dual device might be unpredictable and connect either in LE mode or BR/EDR?